src/framework/ui/modal/modal.component.tsx

```diff
@@ -111,154 +94,100 @@ export type ModalProps = ViewProps & ComponentProps;
 export class Modal extends React.Component<ModalProps> {
 
   static defaultProps: Partial<ModalProps> = {
-    visible: false,
-    isBackDropAllowed: false,
-    animationType: 'none',
-    animationDuration: 300,
+    closeOnBackdrop: true,
   };
 
-  private animation: Animated.Value;
-
-  constructor(props: ModalProps) {
-    super(props);
-
-    this.setAnimation();
-  }
-
-  public componentDidMount(): void {
-    this.startAnimation();
-  }
+  private contentHeight: number = 0;
+  private contentWidth: number = 0;
+  private id: string = '';
 
   public componentWillReceiveProps(nextProps: Readonly<ModalProps>): void {
```

Comment:
`nextProps: ModalProps`. Readonly is unuseful here, but ts plugin generates it everywhere for any reason

Reply:
Done.